src/main/java/com/cleanroommc/groovyscript/compat/mods/pyrotech/StoneOven.java

```diff
@@ -0,0 +1,107 @@
+        super(ModuleTechMachine.Registries.STONE_OVEN_RECIPES);
+    }
+
+    @RecipeBuilderDescription(example = @Example(".input(item('minecraft:diamond')).output(item('minecraft:emerald')).duration(400).name('diamond_campfire_to_emerald_stone')"))
+    public RecipeBuilder recipeBuilder() {
+        return new RecipeBuilder();
+    }
+
+    @MethodDescription(type = MethodDescription.Type.ADDITION, example = @Example("'apple_to_dirt_stone', item('minecraft:apple'), item('minecraft:dirt'), 1000"))
+    public StoneOvenRecipe add(String name, IIngredient input, ItemStack output, int duration) {
+        return recipeBuilder()
+                .duration(duration)
+                .name(name)
+                .input(input)
+                .output(output)
+                .register();
+    }
+
+    @MethodDescription(example = @Example("item('minecraft:porkchop')"))
+    public void removeByInput(ItemStack input) {
+        if (GroovyLog.msg("Error removing stone oven recipe")
+                .add(IngredientHelper.isEmpty(input), () -> "Input 1 must not be empty")
+                .error()
+                .postIfNotEmpty()) {
+            return;
+        }
+        for (StoneOvenRecipe recipe : getRegistry()) {
+            if (recipe.getInput().test(input)) {
+                remove(recipe);
+            }
+        }
+    }
+
+    @MethodDescription(example = @Example("item('minecraft:cooked_porkchop')"))
+    public void removeByOutput(IIngredient output) {
+        if (GroovyLog.msg("Error removing stone oven recipe")
+                .add(IngredientHelper.isEmpty(output), () -> "Output 1 must not be empty")
+                .error()
+                .postIfNotEmpty()) {
+            return;
+        }
+        for (StoneOvenRecipe recipe : getRegistry()) {
+            if (output.test(recipe.getOutput())) {
+                remove(recipe);
+            }
+        }
+    }
+
+    @Property(property = "input", valid = @Comp("1"))
+    @Property(property = "output", valid = @Comp("1"))
+    @Property(property = "name")
+    public static class RecipeBuilder extends AbstractRecipeBuilder<StoneOvenRecipe> {
+
+        @Property(valid = @Comp(type = Comp.Type.GTE, value = "1"))
+        private int duration;
+
+        @RecipeBuilderMethodDescription
+        public RecipeBuilder duration(int time) {
+            this.duration = time;
+            return this;
+        }
+
+        @Override
+        public String getErrorMsg() {
+            return "Error adding Pyrotech Stone Oven Recipe";
+        }
+
+        @Override
+        public void validate(GroovyLog.Msg msg) {
+            validateItems(msg, 1, 1, 1, 1);
+            msg.add(duration < 0, "duration must be a non negative integer, yet it was {}", duration);
+            msg.add(super.name == null, "name cannot be null.");
+            msg.add(ModuleTechBasic.Registries.CAMPFIRE_RECIPE.getValue(super.name) != null, "tried to register {}, but it already exists.", super.name);
```

Comment:
is checking `ModuleTechBasic.Registries.CAMPFIRE_RECIPE` correct? seems like it ought to be `ModuleTechMachine.Registries.STONE_OVEN_RECIPES`